lib/nfsbase.pm

```diff
@@ -0,0 +1,146 @@
+package nfsbase;
+use mmapi;
+use testapi;
+use strict;
+use warnings;
+use utils;
+use mm_network;
+use version_utils;
+
+
+our @ISA    = qw(Exporter);
+our @EXPORT = qw(server_configure_network try_nfsv2 prepare_exports yast_handle_firewall add_rw_share mount_export client_common_tests);
+
+sub server_configure_network {
+    my ($self) = @_;
+
+    # Configure static IP for client/server test
+    configure_default_gateway;
+    configure_static_ip('10.0.2.101/24');
+    configure_static_dns(get_host_resolv_conf());
+
+    if (is_sle('15+')) {
+        record_soft_failure 'boo#1130093 No firewalld service for nfs-kernel-server';
+        systemctl 'stop ' . $self->firewall;
+        systemctl 'disable ' . $self->firewall;
```

Comment:
I am not sure about that but i dont think you need $self here. Can you double check?

Reply:
I don't think it's possible because we use either `SUSEFirewall2` or `firewalld` and `$self->firewall` points to the right one.

Reply:
generally you are right , but not when you are inside `if (is_sle('15+'))` this clearly left you with firewalld only

Reply:
but I don't see the problem if you leave it like this . it is still correct